mono/utils/mono-icall.h

```diff
@@ -0,0 +1,20 @@
+/**
+ * \file
+ */
+
+#ifndef __UTILS_MONO_ICALL_H__
+#define __UTILS_MONO_ICALL_H__
+
+#include <mono/utils/mono-publib.h>
+
+#ifdef ENABLE_ICALL_EXPORT
```

Comment:
Unfortunately, icalls have nothing to do in `utils/` since they are here to support managed code, and in `utils/` we tend to avoid things that support managed code directly. I understand you did that because some code in `utils/` depends on `ICALL_EXPORT`.

Move this file into `metadata/icalls.h`, and move the icalls that depends on `ICALL_EXPORT` to `metadata/`. Following are the only icalls I could identify that are in `utils/` and that needs to be moved to `metadata/`:
 - `ves_icall_System_Threading_Thread_SystemMaxStackSize`: move this function to `threads.c`/`thread-types.h` and create a function `mono_thread_info_get_system_max_stack_size` which is declared in `mono-threads.h` (in place of the icall) and implemented in `mono-threads-{posix,...}.c`
 - `mono_threads_platform_yield`: add a `ves_icall_System_Threading_Thread_Yield` function  in `threads.c`/`thread-types.h` that simply calls `mono_threads_platform_yield`.

Reply:
I believe mini also has icalls, so utils is shared place for it.

Reply:
`mini/` can depend on `metadata/` and `utils`, `metadata/` can depend on `utils/` only, `utils/` cannot depend on anything else. That's excluding other directories, but you get the idea. So `utils/` is not the shared place for that, and it has to be moved to `metadata/`

Reply:
I was unsure of the mini vs. metadata layering. Great, that is easy then, as stated.